src/hotspot/share/services/memReporter.cpp

```diff
@@ -195,6 +201,10 @@ void MemSummaryReporter::report_summary_of_type(MEMFLAGS flag,
     const char*   scale = current_scale();
     out->print("-%26s (", NMTUtil::flag_to_name(flag));
     print_total(reserved_amount, committed_amount);
+    if (flag == mtClassShared) {
+      output()->print(", readonly=" SIZE_FORMAT "%s",
+                      amount_in_current_scale(read_only_bytes), scale);
+    }
```

Comment:
Two days ago, I made a request for change in here but that seems to have disappeared :-(

The read_only_bytes is relevant to mtClassShared only, so the code from lines 174-178 should be moved inside this "if" block.